Makefile

```diff
@@ -0,0 +1,76 @@
+OBJS := main.o Player.o Character.o JSONParser.o 
+CFLAGS := -std=c++17 -Wall -Werror -g 
+CC := g++
+UNIT1 := units/unit1.json
+UNIT2 := units/unit2.json
+UNIT3 := units/unit3.json
+BAD_UNIT := unit_testing/missing_comma.json
+FSYS_FLAG := -lstdc++fs
+OUTBIN := a.out
+
+build: $(OBJS)
+	$(CC) $(CFLAGS) -o $(OUTBIN) $(OBJS) $(FSYS_FLAG)
+
+main.o: main.cpp
+	$(CC) $(CFLAGS) -c main.cpp Character.h Player.h JSONParser.h $(FSYS_FLAG)
+
+JSONParser.o: JSONParser.cpp JSONParser.h
+	$(CC) $(CFLAGS) -c JSONParser.cpp JSONParser.h Character.h $(FSYS_FLAG)
+
+Player.o: Player.cpp Player.h
+	$(CC) $(CFLAGS) -c Player.cpp Player.h Character.h $(FSYS_FLAG)
+
+Character.o: Character.cpp Character.h
+	$(CC) $(CFLAGS) -c Character.cpp Character.h JSONParser.h $(FSYS_FLAG)
```

Comment:
Dependencyben ott kellene lennie annak a h-nak, de  a g++-nak miert is adod at?!

Reply:
@hegyhati dependencynél minden olyan header szerepeljen, amivel utána compileolok? A g++-nak a JSONParser.h vagy a Character headert felesleges átadnom? Nem minden fájlt kell fordítani és buildelni, amit includeolok a .cpp forrásfájlban, hogy a linker össze tudja kapcsolni őket, elkerülve az `undefined reference` és `inclusion error` hibákat?

Reply:
Nem releváns már a kérdésem.

Reply:
Nem is teljesen ertettem :-D